### ISSUE TITLE
TypeError: session.keepAlive is not a function

### ISSUE THREAD

**Issue by jfeigel** [googleapis/nodejs-spanner#26]
#### Environment details

  - OS: macOS Sierra v10.12.6
  - Node.js version: 7.10.0
  - npm version: 4.2.0
  - @google-cloud/spanner version: 0.7.1

#### Steps to reproduce

  1. Open a connection to the database
  2. Let the connection sit for a while

Here is the error we are seeing in Stackdriver:

```
TypeError: session.keepAlive is not a function
    at SessionEvictor.evict (/usr/src/gMail/node_modules/@google-cloud/spanner/src/session-pool.js:608:13)
    at Pool._evict (/usr/src/gMail/node_modules/generic-pool/lib/Pool.js:338:41)
    at Timeout._scheduledEviction.setTimeout [as _onTimeout] (/usr/src/gMail/node_modules/generic-pool/lib/Pool.js:353:14)
    at ontimeout (timers.js:386:14)
    at tryOnTimeout (timers.js:250:5)
    at Timer.listOnTimeout (timers.js:214:5)
```

We specify the keepAlive as 30 in the configuration:

```JavaScript
var Spanner = require('@google-cloud/spanner');

...

var spanner = Spanner(_.pick(config, ['projectId', 'keyFilename']));
var instance = spanner.instance(config.instance);
self.connection = instance.database(config.database, {
  min: 1,
  keepAlive: 30
});
```

**Comment by callmehiphop** [comment]
hey @jfeigel, I tried to reproduce this locally but was unable to do so. My first suggestion would be to try upgrading to `0.8.0`. If that does not resolve your issue, would you mind putting together a gist/repo that consistently reproduces the bug?

**Comment by jfeigel** [comment]
@callmehiphop We solved the issue. Turns out the DBs that we were trying to connect to did not exist, however spanner allowed us to "connect" and didn't send back any kind of error or warning to this fact. So when the session tried the keepAlive it errored out because there wasn't one.

**Comment by vkedia** [comment]
If the DB did not exist, then creating a session itself would fail and it should not even try to keep alive the session. @callmehiphop I think we should fail more gracefully here.

**Comment by callmehiphop** [comment]
Yeah, I agree. I think we might want to either create an `open` method which will report any initial failures creating initial sessions/etc. Or else turn the database object into an event emitter that will emit these types of errors.

@stephenplusplus WDYT?

**Comment by stephenplusplus** [comment]
I would lean towards emitting errors. What do you think, @jfeigel?

**Comment by jfeigel** [comment]
@stephenplusplus Yes, I think emitting errors would be preferable.

**Comment by callmehiphop** [comment]
@jfeigel I've just published a new release (`0.9.0`) that turns database objects into event emitters, you should be able to do the following now.

```js
database.on('error', err => console.error(err));
```

Going to close this now, thanks for the help!
